datadog_checks_base/datadog_checks/utils/decorator.py

```diff
@@ -0,0 +1,17 @@
+# (C) Datadog, Inc. 2018
+# All rights reserved
+# Licensed under Simplified BSD License (see LICENSE)
+
+from datadog_checks.config import is_affirmative
+from ddtrace import tracer, patch
+
+patch(requests=True)
+
+def trace_func(func):
+    def function_wrapper(*args, **kwargs):
+        if is_affirmative(args[1].get('trace_check', False)):
+            with tracer.trace(func.__name__, service=args[0].__class__.__name__):
```

Comment:
This IMO should be:
```
tracer.trace('integration.check', service='IntegrationsTracing', resource=self.name)
```
This would allow us to have a main entry in the trace UI showing `IntegrationsTracing`, then the list of traces broken down by `check.name`. Each one would show (for now, or by default...) one traced function, `integration.check`.

Reply:
My reasoning for making the Service name be the name of the class is so that we aren't locked into only decorating the `check` function. 

If we make the service name static then the set of resources under that are going to be difficult to sift through if we add the decorator to a couple functions in a check.

Reply:
Not sure how the two things are connected - you can have one big service and a bunch of resources grouped by it. Each resource would have one or more spans (likely one per function?) attached to it.

Let's not assume one trace == one function